src/app/shared/helpers/errors.ts

```diff
@@ -6,7 +6,7 @@ interface WithErrorTag { errorTag: string }
  * Add a tag to an error object so it can be recognized later (using the errorHasTag function)
  */
 export function tagError<T>(error: T, tag: string): T & WithErrorTag {
-  return { ...error, errorTag: tag };
+  return Object.assign(error, { errorTag: tag });
```

Comment:
that's not equivalent?

Reply:
Nope, `Object.assign` preserves instances while the spread operator converts any form of data to a plain JS object

Reply:
Ok, makes sense. This (my initial) trick looks a bit a dirty actually. If we are extending an object, we should use inheritance.